stacks-signer/src/chainstate.rs

```diff
@@ -13,17 +13,45 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+use std::time::Duration;
+
 use blockstack_lib::chainstate::nakamoto::NakamotoBlock;
 use blockstack_lib::chainstate::stacks::TenureChangePayload;
 use blockstack_lib::net::api::getsortition::SortitionInfo;
+use blockstack_lib::util_lib::db::Error as DBError;
+use clarity::types::chainstate::BurnchainHeaderHash;
 use slog::{slog_info, slog_warn};
 use stacks_common::types::chainstate::{ConsensusHash, StacksPublicKey};
 use stacks_common::util::hash::Hash160;
 use stacks_common::{info, warn};
 
 use crate::client::{ClientError, StacksClient};
+use crate::config::SignerConfig;
 use crate::signerdb::SignerDb;
 
+#[derive(thiserror::Error, Debug)]
+/// Error type for the signer chainstate module
+pub enum SignerChainstateError {
+    /// Error resulting from database interactions
+    #[error("Database error: {0}")]
+    DBError(DBError),
+    /// Error resulting from crate::client interactions
+    #[error("Client error: {0}")]
+    ClientError(ClientError),
+}
+
+impl From<ClientError> for SignerChainstateError {
```

Comment:
You can remove these implementations and change the enum above like so 

```rust
#[derive(thiserror::Error, Debug)]
/// Error type for the signer chainstate module
pub enum SignerChainstateError {
    /// Error resulting from database interactions
    #[error("Database error: {0}")]
    DBError(#[from] DBError),
    /// Error resulting from crate::client interactions
    #[error("Client error: {0}")]
    ClientError(#[from] ClientError),
}
```

Reply:
Nice!

Reply:
Nice, updated in [62aac67](https://github.com/stacks-network/stacks-core/pull/4956/commits/62aac67851b4104d2f12a549afaa00bbc82436c6)